File: javascript/claude/p80.js
// p80.js - read line without newline (up to 1023 chars)
const readline = require('readline');
const rl = readline.createInterface({ input: process.stdin });

rl.on('line', (line) => {
  const truncated = line.slice(0, 1023);
  console.log(`Read ${truncated.length} char(s): ${truncated}`);
  rl.close();
});

File: java/claude/p80.java
import java.io.*;

// p80.java - read line without newline (truncate to 1023 chars)
public class p80 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        System.out.println("Enter a string:");
        String line = br.readLine();
        if (line != null) {
            String truncated = line.length() > 1023 ? line.substring(0, 1023) : line;
            System.out.println("Read " + truncated.length() + " char(s): " + truncated);
        }
    }
}
